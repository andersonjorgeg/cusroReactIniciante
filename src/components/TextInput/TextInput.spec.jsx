import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { TextInput } from '.';

describe('<TextInput />', () => {
  // isto. deve ter um valor de searchValue
  it('should have a value of searchValue', () => {
    const fn = jest.fn();
    render(<TextInput handleChange={fn} searchValue={'testando'} />);

    const input = screen.getByPlaceholderText(/search by title/i);
    expect(input.value).toBe('testando');
  });

  // isto. deve chamar a função handleChange em cada tecla pressionada
  it('should call handleChange function on each key pressed', () => {
    const fn = jest.fn();
    render(<TextInput handleChange={fn} searchValue={'o valor'} />);

    const input = screen.getByPlaceholderText(/search by title/i);

    const value = 'o valor';

    userEvent.type(input, value);

    expect(input.value).toBe(value);
    expect(fn).toHaveBeenCalledTimes(value.length);
  });
  it('should match snapshot', () => {
    const fn = jest.fn();
    const { container } = render(<TextInput handleChange={fn} searchValue={''} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
