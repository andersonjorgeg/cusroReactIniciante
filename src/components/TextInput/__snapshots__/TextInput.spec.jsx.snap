// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should match snapshot 1`] = `
<input
  class="text-input"
  placeholder="search by title"
  type="search"
  value=""
/>
`;
